catalog/spatial/registry/registry-admin-modules/registry-admin-local-ui/src/main/webapp/js/model/Association.js

```diff
@@ -142,15 +143,17 @@ define([
         },
         removeSegment: function (segment) {
             var model = this;
-            var updatedAssociations = [];
+            var removedAssociations = [];
             var associations = this.get('associations').models;
             var segId = segment.get('segmentId');
             _.each(associations, function (association) {
-                if (association.get('sourceId') !== segId && association.get('targetId') !== segId) {
-                    updatedAssociations.push(association);
+                if (association.get('sourceId') === segId || association.get('targetId') === segId) {
+                    removedAssociations.push(association);
                 }
             });
-            this.set('associations', updatedAssociations);
+            _.each(removedAssociations, function(association){
+               model.get('associations').remove(association);
+            });
 
             var seg = _.find(this.get('associationSegments').models, function (seg) {
```

Comment:
They correct way to do this is to use the underscore methods that have already been mixed-in to backbone. `this.get('associationSegments').find(function(...) ...)` then you don't need to expose their inner api to your code.